sdk/go/core/ydk/path/path.go

```diff
@@ -253,25 +262,57 @@ func getTopEntityFromFilter(filter types.Entity) types.Entity {
 // ReadDatanode populates entity by reading the top level entity from a given data node.
 // Returns the top entity (types.Entity) from readDataNode.
 func ReadDatanode(filter types.Entity, readDataNode types.DataNode) types.Entity {
+
+    ec := types.NewEntityCollection()
+
 	if readDataNode.Private == nil {
-		return nil
+		ydk.YLogError("path.ReadDatanode: The readDataNode is nil; returning empty EntityCollection")
+		return ec
 	}
 
-	topEntity := getTopEntityFromFilter(filter)
-	segmentPath := topEntity.GetEntityData().SegmentPath
-	ydk.YLogDebug(
-		fmt.Sprintf("Reading top entity: '%s'", segmentPath))
-
 	cchildren := C.DataNodeGetChildren(readDataNode.Private.(C.DataNode))
 
 	if cchildren.count == C.int(0) {
-		return topEntity
+		return filter;
 	}
-
+	
```

Comment:
Extra indentation should be deleted

Reply:
Will check-in with next commit due to code overlapping with project in work.